webhook.go

```diff
@@ -54,15 +54,31 @@ func (m *SecretHubMutator) Mutate(ctx context.Context, obj metav1.Object) (bool,
 		return false, nil
 	}
 
-	version, enabled := pod.Annotations["secrethub"]
+	containersStr, enabled := pod.Annotations["secrethub/mutate"]
 	if !enabled {
 		m.logger.Debugf("Skipping pod %s because it is not annotated with secrethub", pod.Name)
 		return false, nil
 	}
 
+	containers := map[string]struct{}{}
+
+	for _, container := range strings.Split(containersStr, ",") {
```

Comment:
Let's also allow `true`.

Reply:
Then what is the behaviour if you have a container named `true`? I think if we want the easy behaviour of "just mutate all containers", we'd have to stick to seperate annotations:

All containers:
```
secrethub/mutate: true
```

Only `myapp1`:
```
secrethub/mutate: true
secrethub/include: myapp1
```

All containers but `myapp1`:
```
secrethub/mutate: true
secrethub/exclude: myapp1
```

Reply:
Do we need to support naming your container `true` and wanting to enable mutation on just that container? That doesn't sound like a realistic scenario at all and I think we can justify having to change your containers name in that case.

That being said, what's the advantage of allowing `true` to enable on all containers? Although it's nice that there's no dependency on the name anymore, I think depending on the name is quite common in Kubernetes, so name changes will probably not taken up lightly anyways? When mutation is enabled for all containers, you might not think about this when adding a new container to the spec. Do you want mutation to be enabled by default in that case? Do note that all containers that are mutated require the command to be explicitly defined in the containers spec.

Furthermore, do note that we have removed the check for secret references, to make it possible to enable mutation for other use cases (e.g. `secrethub.env` file in a volume), so setting mutation to be enabled by default for any containers that are added to the spec also affects containers that do not use SecretHub.

We could introduce an option `secrethub/mutate: detect` to only enable it when secret references are used.

Given all of the above, I'm leaning towards the simple set up where you opt-in for each container explicitly.

Reply:
> Do we need to support naming your container true and wanting to enable mutation on just that container? That doesn't sound like a realistic scenario at all and I think we can justify having to change your containers name in that case.

I do not agree. It shows that the annotation is being abused. Sure, there is probably no one naming their containers `true`, but even having to discuss this point shows that something is smelly.

The same can be said for `secrethub/mutate: detect`.

> depending on the name is quite common in Kubernetes, so name changes will probably not taken up lightly anyways?

Do you have any examples/references for this? If this is indeed the case, I am totally fine with the current situation.

Reply:
Yeah, some examples:
volume names => volumeMounts.name
secret names => env.valueFrom.SecretKeyRef
secret names => volumes.secret.secretName

Reply:
> even having to discuss this point shows that something is smelly

Yeah, you're right there. It doesn't feel completely right.

Reply:
From the API spec:

> Name of the container specified as a DNS_LABEL. Each container in a pod must have a unique name (DNS_LABEL). Cannot be updated.

Combined with you answers, sounds reasonable to use that as a selector.